backend/packages/Upgrade/src/api/services/ExperimentIncludeService.ts

```diff
@@ -42,25 +46,35 @@ export class ExperimentIncludeService {
+    return this.explicitExperimentIndividualInclusionRepository.insertExplicitExperimentIndividualInclusion(ExplicitExperimentIndividualIncludeDocToSave, logger);
   }
 
-  public async deleteExperimentUser(userId: string, experimentId): Promise<ExplicitExperimentIndividualInclusion | undefined> {
-    const deletedDoc = await this.explicitExperimentIndividualInclusionRepository.deleteById(userId, experimentId);
+  public async deleteExperimentUser(userId: string, experimentId, logger: UpgradeLogger): Promise<ExplicitExperimentIndividualInclusion | undefined> {
+    logger.info({ message: `Delete explicitly included user from the experiment. experimentId: ${experimentId}, userId: ${userId}`});
+    const deletedDoc = await this.explicitExperimentIndividualInclusionRepository.deleteById(userId, experimentId, logger);
     return deletedDoc;
   }
 
-  public getAllExperimentGroups(): Promise<ExplicitExperimentGroupInclusion[]> {
-    return this.explicitExperimentGroupInclusionRepository.findAllGroups();
+  public getAllExperimentGroups(logger: UpgradeLogger): Promise<ExplicitExperimentGroupInclusion[]> {
+    logger.info({ message: `Find all groups who are explicitly included at experiment level`});
+    return this.explicitExperimentGroupInclusionRepository.findAllGroups(logger);
   }
 
-  public getExperimentGroupById(type:string, groupId: string, experimentId: string): Promise<ExplicitExperimentGroupInclusion> {
-    return this.explicitExperimentGroupInclusionRepository.findOneById(type, groupId, experimentId);
+  public getExperimentGroupById(type:string, groupId: string, experimentId: string, logger: UpgradeLogger): Promise<ExplicitExperimentGroupInclusion> {
+    logger.info({ message: `Find the group which is explicitly included for the given experiment. experimentId => ${experimentId}, type => ${type}, groupId => ${groupId}`});
+    return this.explicitExperimentGroupInclusionRepository.findOneById(type, groupId, experimentId, logger);
   }
 
-  public async experimentIncludeGroup(groups: Array<{ groupId: string, type: string }>, experimentId: string): Promise<ExplicitExperimentGroupInclusion[]> {
-    const experiment: Experiment = await this.experimentService.findOne(experimentId);
-    if(!experiment) {
+  public async experimentIncludeGroup(groups: Array<{ groupId: string, type: string }>, experimentId: string, logger: UpgradeLogger): Promise<ExplicitExperimentGroupInclusion[]> {
+    let groupIdInfoForLogger = [];
+    let groupTypeInfoForLogger = [];
+    groups.forEach( group => {
+      groupIdInfoForLogger.push(group.groupId);
+      groupTypeInfoForLogger.push(group.type);
+    });
+    logger.info({ message: `Explicitly include groups for the experiment. experimentId => ${experimentId}, groupIds => ${groupIdInfoForLogger}, types => ${groupTypeInfoForLogger}`});
+    const experiment: Experiment = await this.experimentService.findOne(experimentId, logger);
+    if (!experiment) {
       throw new Error('experiment not found');
```

Comment:
You can add logger.error() as well

Reply:
yes this is also fine, but we do have a whole error object, that should be good.

Reply:
code updates, thanks.